hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestExternalPathHandling.java

```diff
@@ -0,0 +1,297 @@
+
+    assertFileGroupCorrectness(instantTime3, partitionPath2, filePath3, fileId3, partitionPath2.isEmpty() ? 2 : 1);
+
+    // clean first commit
+    String cleanTime = HoodieActiveTimeline.createNewInstantTime();
+    HoodieCleanerPlan cleanerPlan = cleanerPlan(new HoodieActionInstant(instantTime2, HoodieTimeline.REPLACE_COMMIT_ACTION, HoodieInstant.State.COMPLETED.name()), instantTime3,
+        Collections.singletonMap(partitionPath1, Collections.singletonList(new HoodieCleanFileInfo(filePath1, false))));
+    metaClient.getActiveTimeline().saveToCleanRequested(new HoodieInstant(HoodieInstant.State.REQUESTED, HoodieTimeline.CLEAN_ACTION, cleanTime),
+        TimelineMetadataUtils.serializeCleanerPlan(cleanerPlan));
+    HoodieInstant inflightClean = metaClient.getActiveTimeline().transitionCleanRequestedToInflight(
+        new HoodieInstant(HoodieInstant.State.REQUESTED, HoodieTimeline.CLEAN_ACTION, cleanTime), Option.empty());
+    List<HoodieCleanStat> cleanStats = Collections.singletonList(createCleanStat(partitionPath1, Arrays.asList(filePath1), instantTime2, instantTime3));
+    HoodieCleanMetadata cleanMetadata = CleanerUtils.convertCleanMetadata(
+        cleanTime,
+        Option.empty(),
+        cleanStats);
+    HoodieTableMetadataWriter hoodieTableMetadataWriter = (HoodieTableMetadataWriter) writeClient.initTable(WriteOperationType.UPSERT, Option.of(cleanTime)).getMetadataWriter(cleanTime).get();
+    hoodieTableMetadataWriter.update(cleanMetadata, cleanTime);
+    metaClient.getActiveTimeline().transitionCleanInflightToComplete(inflightClean,
+        TimelineMetadataUtils.serializeCleanMetadata(cleanMetadata));
+    // make sure we still get the same results as before
+    assertFileGroupCorrectness(instantTime2, partitionPath1, filePath2, fileId2, partitionPath2.isEmpty() ? 2 : 1);
+    assertFileGroupCorrectness(instantTime3, partitionPath2, filePath3, fileId3, partitionPath2.isEmpty() ? 2 : 1);
+
+    // trigger archiver manually
+    writeClient.archive();
+    // assert commit was archived
+    Assertions.assertEquals(1, metaClient.getArchivedTimeline().reload().filterCompletedInstants().countInstants());
+    // make sure we still get the same results as before
+    assertFileGroupCorrectness(instantTime2, partitionPath1, filePath2, fileId2, partitionPath2.isEmpty() ? 2 : 1);
+    assertFileGroupCorrectness(instantTime3, partitionPath2, filePath3, fileId3, partitionPath2.isEmpty() ? 2 : 1);
+
+    // assert that column stats are correct
+    HoodieBackedTableMetadata hoodieBackedTableMetadata = new HoodieBackedTableMetadata(context, writeConfig.getMetadataConfig(), writeConfig.getBasePath(), true);
+    assertEmptyColStats(hoodieBackedTableMetadata, partitionPath1, fileName1);
+    assertColStats(hoodieBackedTableMetadata, partitionPath1, fileName2);
+    assertColStats(hoodieBackedTableMetadata, partitionPath2, fileName3);
+  }
+
+  static Stream<Arguments> getArgs() {
```

Comment:
I had cases for the traditional hudi paths locally for initial test setup validation but removed them as they don't seem relevant after that initial validation that the test itself was set up correctly. Let me know if you would like me to add them back in.

Reply:
should be fine